src/tracer.js

```diff
@@ -218,53 +226,39 @@ export default class Tracer {
       }
     }
 
-    let spanKindValue = userTags[opentracing_tags.SPAN_KIND];
-    let rpcServer = spanKindValue === opentracing_tags.SPAN_KIND_RPC_SERVER;
-
-    let ctx: SpanContext = new SpanContext();
+    let ctx: SpanContext;
     let internalTags: any = {};
+    let hasValidParent = false;
     if (!parent || !parent.isValid) {
       let randomId = Utils.getRandom64();
-      let flags = 0;
-      if (this._sampler.isSampled(operationName, internalTags)) {
-        flags |= constants.SAMPLED_MASK;
-      }
-
+      ctx = new SpanContext(randomId, randomId);
       if (parent) {
+        // fake parent, doesn't contain a parent trace-id, but may contain debug-id/baggage
         if (parent.isDebugIDContainerOnly() && this._isDebugAllowed(operationName)) {
-          flags |= constants.SAMPLED_MASK | constants.DEBUG_MASK;
+          ctx._setSampled(true);
+          ctx._setDebug(true);
           internalTags[constants.JAEGER_DEBUG_HEADER] = parent.debugId;
         }
         // baggage that could have been passed via `jaeger-baggage` header
         ctx.baggage = parent.baggage;
       }
-
-      ctx.traceId = randomId;
-      ctx.spanId = randomId;
-      ctx.parentId = null;
```

Comment:
We have a test in the adaptor that explicitly checks that `parentId` is `null` (as opposed to `undefined`) for a newly created span - do you think this behavior is expected and relied upon elsewhere else?

Reply:
we should restore it, probably assign null explicitly in the constructor